assets/app/scripts/controllers/create.js

```diff
@@ -9,54 +9,74 @@
  */
 angular.module('openshiftConsole')
   .controller('CreateController', function ($scope, DataService, $filter, LabelFilter, $location, Logger) {
-    $scope.projectTemplates = {};
-    $scope.openshiftTemplates = {};
+    var projectTemplates;
+    var openshiftTemplates;
 
-    $scope.templatesByTag = {};
+    // Templates with the `instant-apps` tag.
+    $scope.instantApps;
+
+    // All templates from the shared or project namespace that aren't instant apps.
+    // This is displayed in the "Other Templates" section.
+    $scope.otherTemplates;
+
+    // Set to true when shared templates and project templates have finished loading.
+    $scope.templatesLoaded = false;
 
     $scope.sourceURLPattern = /^((ftp|http|https|git):\/\/(\w+:{0,1}[^\s@]*@)|git@)?([^\s@]+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/;
 
     DataService.list("templates", $scope, function(templates) {
-      $scope.projectTemplates = templates.by("metadata.name");
-      templatesByTag();
-      Logger.info("project templates", $scope.projectTemplates);
+      projectTemplates = templates.by("metadata.name");
+      updateTemplates();
+      Logger.info("project templates", projectTemplates);
     });
 
     DataService.list("templates", {namespace: "openshift"}, function(templates) {
-      $scope.openshiftTemplates = templates.by("metadata.name");
-      templatesByTag();
-      Logger.info("openshift templates", $scope.openshiftTemplates);
+      openshiftTemplates = templates.by("metadata.name");
+      updateTemplates();
+      Logger.info("openshift templates", openshiftTemplates);
     });
 
-    var templatesByTag = function() {
-      $scope.templatesByTag = {};
+    function isInstantApp(template) {
+      if (!template.metadata.annotations || !template.metadata.annotations.tags) {
+        return false;
+      }
+
+      var i, tag, tags = template.metadata.annotations.tags.split(",");
+      for (i = 0; i < tags.length; i++) {
+        tag = $.trim(tags[i]);
+        if (tag === "instant-app") {
+          return true;
+        }
+      }
+
+      return false;
+    }
+
+    function updateTemplates() {
+      // Check if we've loaded templates from both the openshift and project namespaces.
+      $scope.templatesLoaded = openshiftTemplates && projectTemplates;
+      $scope.instantApps = {};
+      $scope.otherTemplates = {};
       var fn = function(template) {
-        if (template.metadata.annotations && template.metadata.annotations.tags) {
-          var tags = template.metadata.annotations.tags.split(",");
-          angular.forEach(tags, function(tag){
-            tag = $.trim(tag);
-            // not doing this as a map since we are dealing with things across namespaces that could have collisions on name
-            $scope.templatesByTag[tag] = $scope.templatesByTag[tag] || [];
-            $scope.templatesByTag[tag].push(template);
-          });
+        var uid = $filter('uid')(template);
```

Comment:
actually, aren't `projectTemplates` and `openshiftTemplates` already keyed by uid? should be the second arg to the `forEach` call